Parse/Internal/File/Controller/PFFileStagingController.h

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <Bolts/BFTask.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol PFFileManagerProvider;
+
+@interface PFFileStagingController : NSObject
+
+@property (nonatomic, weak, readonly) id<PFFileManagerProvider> dataSource;
+
+@property (nonatomic, copy, readonly) NSString *stagedFilesDirectoryPath;
+
+///--------------------------------------
+/// @name Init
+///--------------------------------------
+
+- (instancetype)init NS_UNAVAILABLE;
+- (instancetype)initWithDataSource:(id<PFFileManagerProvider>)dataSource NS_DESIGNATED_INITIALIZER;
+
++ (instancetype)stagingControllerWithDataSource:(id<PFFileManagerProvider>)dataSource;
```

Comment:
Nit: Maybe use `controllerWithDataSource:` to be consistent with other controllers?